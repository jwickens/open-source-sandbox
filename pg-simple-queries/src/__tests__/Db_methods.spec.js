/**
 * @flow
 */

const { Db, Table } = require('../')
const path = require('path')

const randomNum = Math.floor(Math.random() * 1000)
const SQLPATH = path.join(__dirname, 'sql')
const OPTIONS = {
  schemaBase: `db_method_tests_${randomNum}`,
  schemaVersion: 0.4,
  sqlDirectory: SQLPATH
}

const db = new Db(OPTIONS)
beforeAll(async () => {
  await db.sync()
})
afterAll(async () => {
  await db.close()
})

describe('Db methods', () => {
  test('getTable() it can get a table', () => {
    const table = db.getTable('test')
    expect(table).toBeInstanceOf(Table)
  })

  test('getTableNames() it can get table names', async () => {
    const tableNames = await db.getTableNames()
    expect(tableNames).toEqual(expect.arrayContaining(['test']))
  })

  test('query() it can query data', async () => {
    const { rows } = await db.query(`select * from test limit 1`)
    expect(rows).toMatchSnapshot()
  })

  test('find() it can find data', async () => {
    const rows = await db.find(`select * from test limit 1`)
    expect(rows[0].id).toBeDefined()
    expect(rows).toMatchSnapshot()
  })

  test('one() it can get one row of data', async () => {
    const row = await db.one(`select * from test limit 1`)
    expect(row.id).toBeDefined()
    expect(row).toMatchSnapshot()
  })
})
