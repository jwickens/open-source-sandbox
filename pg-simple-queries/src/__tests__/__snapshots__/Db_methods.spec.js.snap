// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Db methods find() it can find data 1`] = `
Array [
  anonymous {
    "id": 1,
    "message": null,
    "tags": Array [
      "a",
    ],
    "testName": null,
  },
]
`;

exports[`Db methods one() it can get one row of data 1`] = `
anonymous {
  "id": 1,
  "message": null,
  "tags": Array [
    "a",
  ],
  "testName": null,
}
`;

exports[`Db methods query() it can query data 1`] = `Array []`;
